x-pack/plugins/stack_alerts/public/rule_types/es_query/expression/es_query_expression.tsx

```diff
@@ -252,6 +264,9 @@ export const EsQueryExpression: React.FC<
           onChange={(xjson: string) => {
             setXJson(xjson);
             setParam('esQuery', convertToJson(xjson));
+
+            const runtimeFields = getRuntimeFields();
+            setEsFields(esFields.concat(runtimeFields));
```

Comment:
this will set `esFields` to the concatenated value, and then when the value changes, concat on the updated value, so it keeps adding to it. I think you need to keep the result of `getFields` and use that to concat.

Reply:
Should be resolved in this commit [1e0540b](https://github.com/elastic/kibana/pull/160319/commits/1e0540b676baffa314aaf4467b2b94975783e624)